kohttp/src/main/kotlin/io/github/rybalkinsd/kohttp/client/ClientBuilderImpl.kt

```diff
@@ -80,21 +77,17 @@ internal class ClientBuilderImpl : ClientBuilder {
         set(value) { builder.cache(value) }
         get() = throw UnsupportedOperationException()
 
-    override var socketFactory: SocketFactory
-        set(value) { builder.socketFactory(value) }
-        get() = throw UnsupportedOperationException()
-
-    override var sslSocketFactory: SSLSocketFactory
-        @Suppress("DEPRECATION")
-        set(value) { builder.sslSocketFactory(value) }
-        get() = throw UnsupportedOperationException()
-
-    override var hostnameVerifier: HostnameVerifier
-        set(value) { builder.hostnameVerifier(value) }
-        get() = throw UnsupportedOperationException()
+    override var sslConfig: SslConfig
+        set(value) {
+            if (value.sslSocketFactory != null && value.trustManager != null) {
+                builder.sslSocketFactory(value.sslSocketFactory, value.trustManager)
+            }
 
-    override var certificatePinner: CertificatePinner
-        set(value) { builder.certificatePinner(value) }
+            value.socketFactory.let { if (it != null) builder.socketFactory(it) }
```

Comment:
Could you please double check if `socketFactory` (not `sslSocketFactory`) has an implication outside of SSL context?

Reply:
It is not directly related to SSL indeed. It looks like SslSocketFactory is a subclass of SocketFactory. I moved it out of the SslConfig class and restored it to the initial state